### PR TITLE
[FrameworkBundle] [Security] Remove trans from the security/core in 2.3 & dir loading

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | #16139, #16133 |
| License | MIT |
| Doc PR | - |

See https://github.com/symfony/symfony/pull/16139#discussion_r41238804.

I think the most efficient solution is to remove translations from `Security/Core` in 2.3 only (should not be propagated to newest branches!) and load both folders if they exist.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by ogizanagi** [comment]
Another solution is to load `Security/Core/Resources/Translations` only if the `Security/Core/composer.json` file exists (since 2.4).

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -553,16 +553,16 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
...
+            if (file_exists($translationsDir = dirname($r->getFilename()).'/../../Resources/translations')) {
                 // in Symfony 2.3, translations are located in the symfony/security package
-                $translationsDir = $dir;
+                $dirs[] = $translationsDir;
             }
 
-            $dirs[] = $translationsDir;
+            if (file_exists($translationsDir = dirname($r->getFilename()).'/../Resources/translations')) {
+                // with Symfony 2.4, the Security component was split into several subpackages
+                // and the translations have been moved to the symfony/security-core package
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Actually, this comment is not completely right (we have translations in both packages). And by the way, having duplicated translations directories leads to inconsistent translations (for example, #15955 changed only one of the two translation files).

Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Well, we have them outside of Core for BC reasons (using them is deprecated and they are removed in 3.0)

Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Well, but then removing them from core in Symfony 2.3 is confusing for users imho if they should rely on them again after upgrading.

Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Well, they should be no translation in security-core in 2.3 (they have been moved there in 2.4). the existing ones there in the 2.3 branch are mistakes when switching branch for merges IMO

**Comment by stof** [comment]
You should not load both folders in the translator IMO. this would create more work for the loader by loading security translations twice, while the second one will be overwritten.

What we should have to avoid missing translations in the new location is a unit test checking that both files are in sync in 2.4+ (actually 2.7+ because of maintained branches)

**Comment by ogizanagi** [comment]
Ok. I've just updated the PR. If the `Security/Core/composer.json` file exists, that means the installed version of the `symfony/security-core` is 2.4 since it only exists as a subpackage since 2.4. 
Then, we'll only load `Security/Core/Resources/translations`.

Otherwise, use `Security/Resources/translations`.

**Comment by stof** [comment]
:+1: for this.

@nicolas-grekas be careful when merging. the removal of the 2 translation files should be reverted when propagating the merge to newer branches

**Comment by xabbuh** [comment]
:+1: 

Status: Reviewed

**Comment by nicolas-grekas** [comment]
Thank you @ogizanagi.

**Pull request merged**